### DISCUSSION TITLE
More Random Distributions

### DISCUSSION THREAD

**Discussion by fweth** [josdejong/mathjs#3048] [Ideas]
It would be cool to have a few more random distributions at hand, at least standard normal, but ideally also multivariate normal.  I implemented standard normal and bivariate normal a while ago, unfortunately I forgot about the ideas behind it, but maybe someone finds it useful:

 
```
function stNml() {
  return (
    math.sqrt(-2 * math.log(math.random())) *
    math.cos(2 * math.pi * math.random())
  );
}

function bvNml(vx, vy, cxy) {
  let a =
    (math.sqrt(vx ** 2 * vy ** 2 - vx * vy * cxy ** 2) + vx * vy) /
    (2 * vx * vy);
  return ((snv) => [
    math.sqrt((1 - a) * vx) * snv[0] + math.sqrt(a * vx) * snv[1],
    math.sqrt(a * vy) * snv[0] + math.sqrt((1 - a) * vy) * snv[1],
  ])([stNml(), stNml()]);
}
```

**Comment by josdejong** [comment]
Thanks for bringing this up, it would be very useful indeed to have more functionality about random distributions.

Related: #836, #472

Anyone interested in picking this up? The existing random functions can also use some love.

**Comment by gwhitney** [comment]
As this appears to be a general idea rather than a specific actionable issue, moving to Discussions.

**Comment by kgryte** [comment]
For univariate distributions, Math.js can likely leverage [stdlib](https://github.com/stdlib-js/stdlib/tree/ede9ef830c3e6efcdf0f427a1dc651856c41e6ab/lib/node_modules/%40stdlib/random/base), which includes various PRNGs supporting seeding, state serialization, and re-instantiation. Multivariate distributions are on the roadmap, but we need to finish some loose ends for vectorized ndarray support, which is necessary for the underlying implementations.
